discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -240,6 +239,79 @@ def detach_all_test_disks(self, node_name):
                 tempfile.gettempdir()), "File unexpectedly not in tmp, avoiding deletion to be on the safe side"
             os.remove(source_file)
 
+    def attach_interface(self, node_name, network_xml, target_interface="vnet3"):
+        """
+        Create network and Interface. New interface will be attached to a given node.
+        """
+        network = self.create_network(network_xml)
+        interface_mac = self.add_interface(node_name, network.bridgeName(), target_interface)
+        return network, interface_mac
+
+    def create_network(self, network_xml):
+        """
+        Create a network from a given xml and return libvirt.virNetwork object
+        """
+        logging.info(f"Creating new network: {network_xml}")
+        network = self.libvirt_connection.networkCreateXML(network_xml)
+        if network is None:
+            raise Exception(f"Failed to create network: {network_xml}")
+        active = network.isActive()
+        if active is not 1:
+            self.destroy_network(network)
+            raise Exception(f"Failed to activate network: {network_xml}")
+        logging.info(f"Successfully created and activated network. name: {network.name()}")
+        return network
+
+    def get_network_by_name(self, network_name):
+        """
+        Get network name and return libvirt.virNetwork object
+        """
+        return self.libvirt_connection.networkLookupByName(network_name)
+
+    def destroy_network(self, network):
+        """
+        Destroy network of a given libvirt.virNetwork object
+        """
+        logging.info(f"Destroy network: {network.name()}")
+        network.destroy()
+
+    def add_interface(self, node_name, network_name, target_interface):
+        """
+        Create an interface using given network name, return created interface's mac address
+        """
+        logging.info(f"Creating new interface attached to network: {network_name}, for node: {node_name}")
+        net_leases = self.list_leases(network_name)
+        mac_addresses = []
+        for lease in net_leases:
+            mac_addresses.append(lease['mac'])
+        command = f"virsh attach-interface {node_name} network {network_name} --target {target_interface} --persistent"
+        utils.run_command(command)
+        try:
+            waiting.wait(
+                    lambda: len(self.list_leases(network_name)) > len(mac_addresses),
```

Comment:
In case of running the `add_interface` function in parallel, this would cause a race-condition.
You should verify the `target_interface` was created.

Reply:
The issue is that the taken target is not the actual. the machine set different target, so it make it difficult to understand what interface was added

Reply:
Test will be flaky only if they will use the same network (two different tests use same network). which should not happen.

Reply:
So that's no biggy. Maybe just add a comment about that scenario so it won't be a surprise.

Reply:
added